src/client/Microsoft.Identity.Client/Internal/DeviceCodeResponse.cs

```diff
@@ -32,9 +33,15 @@ internal class DeviceCodeResponse : OAuth2ResponseBase
         public string VerificationUri { get; set; }
 
         [JsonProperty("expires_in")]
+#if SUPPORTS_SYSTEM_TEXT_JSON
```

Comment:
Can you do a similar trick to `JsonProperty` so that we can avoid adding these pragmas everywhere?

Reply:
It would be similar to [JsonObject](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/Platforms/net6/JsonObjectAttribute.cs) which is an empty class on net6 platform, but a proper Json.Net class on non-net6 platforms.
So `JsonNumberHandling` would be opposite - it would be a dummy class on non-net6 platforms. I think it's possible but maybe a bit tricky, since you'd have to add that class on every non-net6 platform.

[JsonIgnore](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/e774d1d9791ce070355dd3aa17709683636de111/src/client/Microsoft.Identity.Client/Internal/JsonWebToken.cs#L250) is similar and I just used `#if` flags.

Up to you whichever way looks cleaner.